examples/motoko_examples/whoami/src/whoami.ts

```diff
@@ -0,0 +1,73 @@
+import {
+    Canister,
+    CanisterResult,
+    ic,
+    Init,
+    PostUpgrade,
+    Principal,
+    Query,
+    UpdateAsync,
+    Variant
+} from 'azle';
+
+type WhoAmICanister = Canister<{
+    installer(): CanisterResult<Principal>;
+    argument(): CanisterResult<Principal>;
+    whoami(): CanisterResult<Principal>;
+    id(): CanisterResult<Principal>;
+    idQuick(): CanisterResult<Principal>;
+}>;
+
+type WhoAmIResult = Variant<{
+    ok: Principal;
+    err: string;
+}>;
+
+let install: Principal;
```

Comment:
These types aren't quite correct if you think about it. I think we should type these as `Opt<Principal>` and set their initial values to null. Right now their initial values are `undefined` so the type should actually be `Principal | undefined` but it isn't.